runelite-client/src/main/java/net/runelite/client/plugins/boosts/BoostsPlugin.java

```diff
@@ -233,18 +233,36 @@ public void onGameTick(GameTick event)
 	{
 		lastTickMillis = System.currentTimeMillis();
 
-		if (config.groupNotifications())
+		if (config.groupNotifications() && !boostedSkillsChanged.isEmpty())
 		{
-			if (boostedSkillsChanged == 1)
+			if (boostedSkillsChanged.size() == 1)
 			{
-				notifier.notify(lastBoostedSkillChanged + " level is getting low!");
-				boostedSkillsChanged = 0;
+				notifier.notify(boostedSkillsChanged.get(0) + " level is getting low!");
 			}
-			else if (boostedSkillsChanged > 1)
+			else
 			{
-				notifier.notify("Multiple skills are getting low!");
-				boostedSkillsChanged = 0;
+				String notification = "";
+				int i = 1;
+				for (final String skill : boostedSkillsChanged)
```

Comment:
This should just be a `for (int i = 0; i < boostedSkillsChanged.size(); i++)` style loop.

Reply:
Updated and retested with the change